File: js/stringMerge.js
function stringMerge(arr1, arr2) {
    merged = []

    while (arr1.length || arr2.length > 0) {
        if (arr1.length > 0) {
            merged.push(arr1.shift())
        }
        if (arr2.length > 0) {
            merged.push(arr2.shift())
        }
    }
    return merged.join('')
}
l1 = ["a", "b", "c"]
l2 = ["f", "d", "s"]
console.log(stringMerge(l1, l2))

File: c#/StringMerge.cs
using System;
using System.Collections.Generic;

namespace ConsoleApp11
{
    class Program
    {
        static string stringMerge(string a, string b)
        {
            List<char> merged = new List<char>();
            int lengthA = a.Length;
            int lengthB = b.Length;
            int i = 0;
            while (lengthA > 0 || lengthB > 0)
            {
                merged.Add(a[i]);
                lengthA--;
                merged.Add(b[i]);
                lengthB--;
                i++;
            }
            return string.Join("", merged.ToArray());
        }
        static void Main(string[] args)
        {
            Console.WriteLine(stringMerge("abs","sfd"));
        }
    }
}
